src/pudl/workspace/zenodo_download.py

```diff
@@ -0,0 +1,339 @@
+        """
+        with open(os.path.expanduser("~/.pudl.yml"), "r") as f:
+            cfg = yaml.load(f.read(), Loader=yaml.FullLoader)
+
+            if "zenodo_download" not in cfg:
+                raise ValueError("No zenodo_download config in ~/.pudl.yml")
+
+        return cfg
+
+    def collect(self, archive, filters=None):
+        """
+        Download the files from the provided archive to the appropriate
+        PUDL_IN directory, coordinating other methods as needed
+
+        Args:
+            archive (str): name of the archive, must be available in the
+                           ~/.pudl.yml
+            filters (dict): limit retrieved files to those where the
+                            datapackage.json["parts"] key & val pairs match
+                            those in the filter
+        Returns:
+            None
+        """
+        if filters is None:
+            filters = {}
+
+        dpkg = self.datapackage_contents(archive)
+
+        output_dir = os.path.join(self.output_root, archive)
+        os.makedirs(output_dir, exist_ok=True)
+
+        self.logger.debug("Datapackage lists %d resources" %
+                          len(dpkg["resources"]))
+
+        for r in dpkg["resources"]:
+
+            if self.passes_filters(r, filters):
+                local_path = self.download_resource(r, output_dir)
+                self.logger.info("Downloaded %s" % local_path)
+
+    def datapackage_contents(self, archive):
+        """
+        Produce the contents of the remote datapackage.json file
+
+        Args:
+            archive (str): name of the archive, must be available in the
+
+        Return:
+            dict representation of the datapackage.json file, as contained
+                within the remote archive
+        """
+        try:
+            doi = self.dois[archive]
+        except KeyError:
+            msg = "No DOI for %s in ~/.pudl.yml" % archive
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        try:
+            dpkg_url = self.doi_to_url(doi)
+        except KeyError:
+            msg = "No datapackage for %s" % doi
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        response = requests.get(dpkg_url, params={"access_token": self.token})
+        jsr = response.json()
+
+        if response.status_code > 299:
+            msg = "Failed to retrieve %s" % dpkg_url
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        files = {x["filename"]: x for x in jsr["files"]}
+
+        response = requests.get(
+            files["datapackage.json"]["links"]["download"],
+            params={"access_token": self.token})
+
+        if response.status_code > 299:
+            msg = "Failed to retrieve datapackage for %s: %s" % (
+                archive, response.text)
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        ymlr = yaml.load(response.text, Loader=yaml.FullLoader)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5554119)